src/commands/tail/mod.rs

```diff
@@ -51,3 +51,79 @@ fn find_open_port(requested: Option<u16>, default: u16) -> Result<u16, failure::
         Ok(socket.local_addr()?.port())
     }
 }
+
+#[cfg(test)]
+mod tests {
+    // These tests are extremely stateful; since what we are testing is how this function behaves
+    // when requested ports are unavailable, and since our tests run concurrently, each test uses
+    // unique ports to avoid collisions. There are two possible solutions to this problem:
+    // 1. require that these tests be run serially, and find a way to clean up bound ports
+    // 2. use dependency injection and write a substitute for the TcpListener::bind fn.
+    use super::*;
+
+    #[test]
+    fn test_find_open_port_no_requested_default_available() {
+        let default = 3000;
+        let port = find_open_port(None, default).unwrap();
```

Comment:
can we name `None` above like we do for default, im not sure what the parameter is before hunting it down in the function definition

Reply:
sure; it's also called out in the test fn name.